File: Lista de Exercícios/Python/5. Polimorfismo/Polimorfismo.py
class Animal:
    def __init__(self, som):
        self.som = som
    
    def somEmitido(self):
        print(f"Som: {self.som}")

class Cachorro(Animal):
    def som(self):
        return "AuAu"

class Gato(Animal):
    def som(self):
        return "Miau"
    
def somEmitido(self):
        print(f"Som: {self.som}")

for animal in [Cachorro(), Gato()]:
    somEmitido(animal)

File: Lista de Exercícios/Go/5. Polimorfismo/Polimorfismo.go
package main

import (
	"fmt"
)

type Animal interface {
	SomEmitido() string
}

type Cachorro struct{}

func (c Cachorro) SomEmitido() string {
	return "AuAu"
}

type Gato struct{}

func (g Gato) SomEmitido() string {
	return "Miau"
}

func main() {

	animais := []Animal{Cachorro{}, Gato{}}

	for _, animal := range animais {
		fmt.Printf("Som: %s\n", animal.SomEmitido())
	}
}
